index.html

```diff
@@ -179,12 +179,22 @@
           "upgrade-2": L.icon({iconUrl: 'imgs/marker-icon-2x-violet.png', ...iconData}),
           "mbta-token": L.icon({iconUrl: 'imgs/marker-icon-2x-orange.png', ...iconData})
         };
-        for (const {x, y, msg, type} of window.objectsData) {
-          L.marker([-y, x], {
-            icon: objIcons[type],
-            title: msg
-          }).addTo(map).on('click', ()=>alert("type: " + type + "\nmsg: " + msg));
-        }
+
+        // split markers into separate LayerGroups
+        const markerLayers = window.objectsData.reduce((result, obj) => {
+          if (!result[obj.type]) {
+            result[obj.type] = L.layerGroup().addTo(map);
+          } else {
```

Comment:
There shouldn't be an else here, otherwise we miss making a marker for the first object of each type